modules/standard/Random.chpl

```diff
@@ -638,6 +638,23 @@ module Random {
         const oddseed = if seed % 2 == 0 then seed + 1 else seed;
         return oddseed;
       }
+      /*
+      Generate a seed based on the entropy pool in the OS; this requires
+      the presence of /dev/urandom.  This should be standard on both macOS
+      and Linux systems.
+
+      For highly parallel applications on one locale, it may be necessary
```

Comment:
Can you add a test that demonstrates either A) that this is a problem, or B) an example of locking to work around the problem?

Reply:
B is preferable, as A is a race condition and so a little difficult to reproduce (I think I had to use 48 cores to reproduce it beforehand) but I can see if I can create a reproducer.

Reply:
I wouldn't use locking but would rather have each thread read from a different offset in /dev/urandom. However I think it is surprising that two reads (on the same or different thread) would get the same 8 bytes in any situation. Perhaps this is a situation where O_DIRECT would be useful.

Reply:
Another thing worth considering is calling `getrandom` on linux, which might avoid a thread-safety issue (in the event that multiple threads read at the same time, I wonder if the file page cache is caching the random data).

Reply:
I was surprised by it as well, but maybe you're right about the file page cache.  I was using 128 random bits to generate UUIDs, and was getting errors resulting from unexpected collisions (as two objects with the same name were created but one overwrote the other and so it had undefined state).  

My understanding regarding offsets is that the channel object is thread safe, right?  So if I'm accessing the entropy pool via the same channel object, that should return different data.  Now that I think about it, it's possible I was actually opening two separate channels to /dev/urandom, which might explain why there was some funky race condition magic happening.

Re: getrandom, that's an excellent idea.  I assume we'd just have to call it from C and use the results.  For a seed, casting to an int is fine.  But as @LouisJenkinsCS brought up, perhaps we'd want to place the /dev/urandom functionality into a separate place in the random module so that we can use it for other purposes (for instance, getting in 128 bits of random data) and then just call into that for the seed generator.

Reply:
I think it's already easy to use `/dev/urandom` - you just have to read from a file - what's the big deal?

Reply:
I agree that it's easy to use /dev/urandom.  I was just responding to your suggestion that we potentially use getrandom, instead.  If it turns out that the threading is a problem and something like getrandom dodges issues around it, then making that solution built-in as opposed to something a user would have to roll out could be convenient, is all.

Reply:
Yeah. I think we need a reproducer for the duplicate-random-values issue so we can evaluate what it takes to fix.